modules/@apostrophecms/schema/index.js

```diff
@@ -611,92 +616,207 @@ module.exports = {
+          req,
+          schema,
+          convertErrors,
+          destination,
+          nonVisibleFields
+        });
+
+        for (const error of validErrors) {
+          self.apos.util.error(error.stack);
+        }
+
+        if (validErrors.length) {
+          throw validErrors;
+        }
+      },
+
+      async getNonVisibleFields({
+        req, schema, destination, nonVisibleFields = new Set(), fieldPath = ''
+      }) {
+        for (const field of schema) {
+          const curPath = fieldPath ? `${fieldPath}.${field.name}` : field.name;
+          const isVisible = await self.isVisible(req, schema, destination, field.name);
+          if (!isVisible) {
+            nonVisibleFields.add(curPath);
+            continue;
+          }
+          if (!field.schema) {
+            continue;
+          }
+
+          // Relationship does not support conditional fields right now
+          if ([ 'array' /*, 'relationship' */].includes(field.type) && field.schema) {
+            for (const arrayItem of destination[field.name] || []) {
+              await self.getNonVisibleFields({
+                req,
+                schema: field.schema,
+                destination: arrayItem,
+                nonVisibleFields,
+                fieldPath: `${curPath}.${arrayItem._id}`
+              });
             }
-            if (isParentVisible === false) {
+          } else if (field.type === 'object') {
+            await self.getNonVisibleFields({
+              req,
+              schema: field.schema,
+              destination: destination[field.name],
+              nonVisibleFields,
+              fieldPath: curPath
+            });
+          }
+        }
+
+        return nonVisibleFields;
+      },
+
+      async handleConvertErrors({
+        req,
+        schema,
+        convertErrors,
+        nonVisibleFields,
+        destination,
+        destinationPath = '',
+        hiddenAncestors = false
+      }) {
+        const validErrors = [];
+        for (const error of convertErrors) {
+          const [ destId, destPath ] = error.path.includes('.')
+            ? error.path.split('.')
+            : [ null, error.path ];
+
+          const curDestination = destId
+            ? destination.find(({ _id }) => _id === destId)
+            : destination;
+
+          const errorPath = destinationPath
+            ? `${destinationPath}.${error.path}`
+            : error.path;
+
+          // Case were this error field hasn't been treated
+          // Should check if path starts with, because parent can be invisible
+          const nonVisibleField = hiddenAncestors || nonVisibleFields.has(errorPath);
+
+          // We set default values only on final error fields
+          if (nonVisibleField && !error.data?.errors) {
+            const curSchema = self.getFieldLevelSchema(schema, error.schemaPath);
+            self.setDefaultToInvisibleField(curDestination, curSchema, error.path);
```

Comment:
I'm not sure this method is really useful.
I wrote a test for it and realized I saw it work only for a string that has a pattern.
In the case of in integer having a `min` or a `max` for example, the convert updates the value itself, it does not need this method.
Maybe there are some cases I didn't test.